### PR TITLE
ESLint: Simplify via configuration cascading

### PR DESCRIPTION
**Pull request by jsnmoon**
### Changes proposed in this Pull Request:
Simplify ESLint configuration by cascading our configuration files. This significantly simplifies the `yarn lint` command.

See https://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

No, this does not add/remove features from Jetpack.

#### Testing instructions:

Run `yarn lint` and ensure that it outputs the same results as the current `master` branch.

#### Proposed changelog entry for your changes:

No changelog necessary.

### CODE REVIEW

**Review by jeherve — DISMISSED** [review]
This seems to work well in my tests. 👍

File: package.json [hunk 1]
```diff
@@ -52,9 +52,7 @@
 		"docker:clean": "yarn docker:compose down --rmi all -v && rm -rf docker/wordpress/* docker/wordpress/.htaccess docker/wordpress-develop/* docker/logs/* docker/data/mysql/*",
 		"docker:env": "node -e \"require('fs').createWriteStream( 'docker/.env', { flags: 'a' } );\"",
 		"docker:update-core-unit-tests": "yarn docker:compose exec wordpress svn up /tmp/wordpress-develop/tests/phpunit/data/ /tmp/wordpress-develop/tests/phpunit/includes",
-		"lint": "yarn lint-es6 && yarn lint-es5",
-		"lint-es5": "eslint _inc/*.js modules -c modules/.eslintrc.js",
-		"lint-es6": "eslint --ext .js --ext .jsx ./*.js _inc/client extensions -c .eslintrc.js",
+		"lint": "eslint --ext .js,.jsx ./*.js _inc/*.js _inc/client extensions modules",
```
Comment (simison) [inline package.json]: Curious why `./*.js _inc/*.js` here if we already have `--ext`?

Should this be something like `. _inc extensions modules`?

Comment (simison) [reply package.json]: Suppose root level folder covers sub-folders? i.e. no need `_inc/*/**`

**Comment by simison** [comment]
Personally, I find `overrides` section over individual files for _much_ better readability and grokking what actually is going on with linting in the repository.

That said, I appreciate that it might be impossible to organize linting that way in Jetpack because of such crazy variety of linting we need to do per-folder basis. :-/

File: package.json (hunk 1 above)
Comment (brbrr) [reply package.json]: I think `--ext` means the list of file types to lint, while `./*.js _inc/*.js` are wildcard file paths.

Comment (simison) [reply package.json]: Docs suggest that both folders and globs work: https://eslint.org/docs/user-guide/command-line-interface (tho seems to be recommended to have them in quotes to ensure they're the same across environments).

At any rate, no action needed in this PR. :-)

File: extensions/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+module.exports = {
```
Comment (simison) [inline extensions/.eslintrc.js]: Please remember to sync this file to wpcom in D31208-code

Comment (simison) [reply extensions/.eslintrc.js]: >Please remember to sync this file to wpcom in D31208-code

Ahh nope, I forgot that we do linting there little differently. :-)

File: package.json (hunk 1 above)
Comment (jsnmoon) [reply package.json]: @simison: `. _inc extensions modules` seems much better for its simplicity. I'll go ahead and make the change.

**Comment by jsnmoon** [comment]
I changed `yarn lint` to include more files and fixed previously unnoticed linting violations in `_inc/lib/tracks/`. 

This is ready for another round of reviews!

**Review by jeherve — APPROVED** [review]
This still works for me. 👍

**Review by simison — APPROVED** [review]

**Comment by simison** [comment]
Just wanted to note that `yarn lint` now brings 100+ lint warnings — would be nice to get those sorted out asap. They seem pretty trivial luckily. :-)

**Comment by jsnmoon** [comment]
Thanks for the quick 👀 and 🚢, everyone!

**Pull request merged**